Thank you for the detailed review and the approval!

### CI Failure Confirmation
I can confirm that the Ubuntu 22.04 linker error (`undefined reference to pthread_create`) is 100% pre-existing and unrelated to this PR. 

Looking at the repository's recent commit history on `main`, every single C++ compilation workflow has been failing with a red "X" ever since the `apache-arrow 24.0.0` dependency upgrade. Specifically, functional C++ commits like #913, #914, #904, and #915 are all hitting this exact environment linker issue, while only isolated markdown/license tasks (like #910) manage to pass.

You can merge now I confirmed.

--- a/cpp/src/graphar/high-level/edges_builder.h
+++ b/cpp/src/graphar/high-level/edges_builder.h
@@ -24,6 +24,7 @@
 #include <map>
 #include <memory>
 #include <string>
+#include <unordered_map>
 #include <utility>
 #include <vector>
 

